boltdesignsystem/bolt
# Typeahead: Add Ability To Customize Data `Keys`

## Jira
N/A

## Summary
Manually ports over the new `keys` prop added to Typeahead from  #1542.

## Details
This adds the ability for any incoming result data to be mapped to Typeahead's internals (ex. wire up data with different key names / nested data) + adds docs and demos to demonstrate how to configure this.

This PR also updates the docs site plumbing to automatically pull in the new set of example JSON data needed in order to properly demo this Typeahead use case (plus reused in a few other demos — separate PR) + auto wire up any Typeahead demo JS files added moving forward.

Furthermore, this PR _also_ updates the [docs site Schema Table](https://github.com/boltdesignsystem/bolt/pull/1564/files#diff-238a1e3a2ba3184cf8431790b5266791) twig file to handle arrays of data so the entire docs site doesn't crash and burn when trying to display the schema prop data for `keys` 🙂  

## How to test
- Review changes
- Confirm the new `customize-data-keys` Typeahead demo works as expected: renders results correctly + properly filters / sorts these results given that they use a different data structure under the hood.

## Dependencies
- Depended on for the incoming Typeahead filter / sort demos (reuse the same JSON data)
- Other Typeahead demos (take advantage of the auto-wiring of demo code)
